onnxruntime/core/util/thread_utils.h

```diff
@@ -25,21 +25,25 @@ struct OrtThreadPoolParams{
   size_t* affinity_vec = nullptr;
   size_t affinity_vec_len = 0;
   const ORTCHAR_T* name = nullptr;
-} ;
+};
 
 struct OrtThreadingOptions {
   // Params for creating the threads that parallelizes execution of an op
   OrtThreadPoolParams intra_op_thread_pool_params;
 
   // Params for creating the threads that parallelizes execution across ops
   OrtThreadPoolParams inter_op_thread_pool_params;
-} ;
+};
 
 namespace onnxruntime {
 
 namespace concurrency {
-
+enum class ThreadPoolType : uint8_t {
+  INTRA_OP,
+  INTER_OP
+};
 std::unique_ptr<ThreadPool> CreateThreadPool(Env* env, OrtThreadPoolParams options,
+                                             ThreadPoolType tpool_type,
```

Comment:
Can we have three functions: 

1. CreateThreadPool
2. CreateInterOpThreadPool, which calls CreateThreadPool
3. CreateIntraOpThreadPool, which calls CreateThreadPool

Sometimes we just want to create a thread pool and run some tests around the thread pool, and don't really care about session or INTRA_OP/INTER_OP

Reply:
Creating too many options creates avenues for mistakes esp. given the nuances around when openmp is enabled and we create the ORT tp only for inter-op. I can make the threadpool type have a default value so that test code doesn't have to worry about the type of pool they're creating. Would that work?

Reply:
Never mind. Let's keep it simple